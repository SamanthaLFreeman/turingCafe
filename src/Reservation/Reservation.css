article {
  background-color: #fff;
  border: solid 6px teal;
  border-radius: 10px;
  color: teal;
  margin: 15px;
  padding: 10px;
  width: 250px;
}
