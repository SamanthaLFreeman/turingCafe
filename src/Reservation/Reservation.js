import React from 'react';

const Reservation = ({reservation}) => {
  return (
    <article>
      <h2>{reservation.name}</h2>
      <p>{reservation.date}</p>
      <p>{reservation.time}</p>
      <p>Number of guests: {reservation.number}</p>
      <button>Cancel</button>
    </article>
  )
}

export default Reservation;