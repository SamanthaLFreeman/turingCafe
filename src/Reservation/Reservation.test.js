import React from 'react';
import { shallow } from 'enzyme';
import Reservation from './Reservation';

describe('Reservation', () => {
  let wrapper;
  const reservationMock = {
    name: 'Sam',
    date: '1/14',
    time: '16:00',
    number: 2
  }
  const cancelReservationMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
    <Reservation 
      reservation={reservationMock}
        cancelReservation={cancelReservationMock} />)
  })

  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call cancelReservation with the current id passed through on click of the button', () => {
    wrapper.find('.cancel-btn').simulate('click');
    expect(cancelReservationMock).toBeCalledWith(reservationMock.id)
  })
})