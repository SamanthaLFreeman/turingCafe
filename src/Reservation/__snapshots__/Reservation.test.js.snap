// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation should match the snapshot with all of the data passed through 1`] = `
<article>
  <h2>
    Sam
  </h2>
  <p>
    1/14
  </p>
  <p>
    16:00
  </p>
  <p>
    Number of guests: 
    2
  </p>
  <button
    className="cancel-btn"
  >
    Cancel
  </button>
</article>
`;
